shm/Ship.cpp

```diff
@@ -33,26 +23,59 @@ Ship& Ship::operator+=(size_t num) {
     return *this;
 }
 
-size_t Ship::getCapacity() const { 
-    return capacity_; 
+size_t Ship::getCapacity() const {
+    return capacity_;
+}
+
+size_t Ship::getMaxCrew() const {
+    return maxCrew_;
 }
 
-size_t Ship::getMaxCrew() const { 
-    return maxCrew_; 
+size_t Ship::getSpeed() const {
+    return speed_;
 }
 
-size_t Ship::getSpeed() const { 
-    return speed_; 
+std::string Ship::getName() const {
+    return name_;
 }
 
-std::string Ship::getName() const { 
-    return name_; 
+size_t Ship::getId() const {
+    return id_;
 }
 
-size_t Ship::getId() const { 
-    return id_; 
+const std::vector<std::unique_ptr<Cargo>>& Ship::getCargoVec() const {
+    return cargoVec_;
 }
 
-const std::vector<std::unique_ptr<Cargo>>& Ship::getCargoVec() const { 
-    return cargoVec_; 
+void Ship::load(std::unique_ptr<Cargo> cargo) {
+    if (!cargo) {
+        return;
+    }
+    for (auto& el : cargoVec_) {
+        if (*cargo == *el) {
+            *el += cargo->getAmount();
+            return;
+        }
+    }
+    cargoVec_.push_back(move(cargo));
 }
+
+void Ship::unload(Cargo* cargo) {
+    if (!cargo) {
+        return;
+    }
+    for (auto& el : cargoVec_) {
```

Comment:
maybe  it will be better to switch "el" in range loop to "cargoOnBoard" - it is more readable